lib/tasks/puppet.rake

```diff
@@ -23,140 +23,6 @@ namespace :puppet do
-          begin
-            ['new', 'updated', 'obsolete'].each { |kind| changes[kind].each_key { |k| changes[kind.to_s][k] = changes[kind.to_s][k].to_json } }
-            errors = PuppetClassImporter.new.obsolete_and_new(changes)
-          rescue => e
-            errors = e.message + "\n" + e.backtrace.join("\n")
-          end
-          if args.batch
-            Rails.logger.warn "Failed to refresh puppet classes: #{errors}"
-          else
-            if errors.empty?
-              puts "Import complete"
-            else
-              puts "Problems were detected during the execution phase"
-              puts
-              puts errors.each { |error| error.gsub(/<br\/>/, "\n") } << "\n"
-              puts
-              puts "Import failed"
-            end
-          end
-        end
-      end
-    end
-
-    desc "Imports only the puppet environments from SmartProxy source."
-    task :environments_only, [:batch] => :environment do |t, args|
-      User.as_anonymous_admin do
-        args.batch = args.batch == "true"
-        puts Foreman::Deprecation.deprecation_warning('1.24', 'use API/CLI commands instead of puppet:import:environments_only rake task')
-        puts " ================================================================ "
-        puts "Import starts: #{Time.now.strftime('%Y-%m-%d %H:%M:%S %Z')}"
-
-        proxies = SmartProxy.with_features("Puppet")
-        if proxies.empty?
-          puts "ERROR: We did not find at least one configured Smart Proxy with the Puppet feature"
-          exit 1
-        end
-        if ENV["proxy"]
-          proxy = proxies.select {|p| p.name == ENV["proxy"]}.first
-          unless proxy.is_a?(SmartProxy)
-            puts "Smart Proxies #{ENV['proxy']} was not found, aborting"
-            exit 1
-          end
-        end
-        proxy ||= proxies.first
-        # Evaluate any changes that exist between the database of environments and the on-disk puppet installation
-        begin
-          puts "Importing environments" unless args.batch
-          importer = PuppetClassImporter.new({ :url => proxy.url })
-          puts "New environments: #{importer.new_environments.join(', ')} "
-          puts "Old environments: #{importer.old_environments.join(', ')} "
-          importer.new_environments.each { |new_environment| Environment.create!(:name => new_environment) }
-          importer.old_environments.each { |old_environment| Environment.find_by_name(old_environment).destroy }
-        rescue => e
-          puts "Problems were detected during the evaluation phase"
-          puts
-          puts e.message.gsub(/<br\/>/, "\n") + "\n"
-          puts
-          puts "Please fix these issues and try again"
-          exit 1
-        end
-        puts "Import ends: #{Time.now.strftime('%Y-%m-%d %H:%M:%S %Z')}"
-        puts " ================================================================ "
-      end
-    end
-  end
 
   namespace :import do
     desc "
```

Comment:
I'm wondering if the part below is still relevant. Looks like the code doesn't handle class parameters so I doubt anyone's using this. Perhaps deprecate it in a separate PR?

Reply:
PRs welcome :wink: